File: app/src/main/java/education/mahmoud/quranyapp/data_layer/local/room/SurahDAO.java
package education.mahmoud.quranyapp.data_layer.local.room;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.Query;
import android.arch.persistence.room.Update;

import java.util.List;

@Dao
public interface SurahDAO {

    @Insert
    public void addSurah(SuraItem suraItem);

    @Update
    public void updateSurah(SuraItem suraItem);

    @Query("select  * from surah ")
    public List<SuraItem> getAllSurah();

    @Query("select name from surah ")
    public List<String> getAllSurahNames();

    @Query("select  * from surah where `index` = :id")
    public SuraItem getSurahByIndex(int id);

    @Query("select  * from surah where `name` = :name")
    public SuraItem getSurahByName(String name);

    @Query("select startIndex from surah where `index` = :index")
    int getSuraStartpage(int index);
}

File: app/src/main/java/education/mahmoud/quranyapp/datalayer/local/room/SurahDAO.kt
package education.mahmoud.quranyapp.datalayer.local.room

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import androidx.room.Update

@Dao
interface SurahDAO {
    @Insert
    fun addSurah(suraItem: SuraItem)

    @Update
    fun updateSurah(suraItem: SuraItem)

    @get:Query("select  * from surah ")
    val allSurah: List<SuraItem>

    @get:Query("select name from surah ")
    val allSurahNames: List<String>

    @Query("select  * from surah where `index` = :id")
    fun getSurahByIndex(id: Int): SuraItem

    @Query("select  * from surah where `name` = :name")
    fun getSurahByName(name: String): SuraItem?

    @Query("select startIndex from surah where `index` = :index")
    fun getSuraStartpage(index: Int): Int

    @Insert
    fun addSurahs(surrahs: List<SuraItem>)
}
